File: Models/Chapter.cs
using System;

namespace coursework_itransition.Models
{
    public class Chapter
    {
        public string ID { get; set; }

        public string CompositionID { get; set; }

        [System.Text.Json.Serialization.JsonIgnore]
        public Composition Composition { get; set; }

        public DateTime CreationDT { get; set; }

        public DateTime LastEditDT { get; set; }

        public string Title { get; set; }
        
        public string Contents { get; set; }

        public int Order { get; set; }

        // public List<string> Likes;
    }
}

File: scripts/Models/chapter.ts
export class Chapter {
    id: string;
    title: string;
    compositionID: string;
    creationDT: string;
    lastEditDT: string;
    order: number;

    static equal(first: Chapter, second: Chapter): boolean{
        if (first.id             != second.id)            return false;
        if (first.title          != second.title)         return false;
        if (first.compositionID  != second.compositionID) return false;
        if (first.creationDT     != second.creationDT)    return false;
        if (first.lastEditDT     != second.lastEditDT)    return false;
        if (first.order          != second.order)         return false;

        return true;
    };
};